staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -242,3 +242,44 @@ func convertJSONSchemaPropsOrStringArray(in *apiextensions.JSONSchemaPropsOrStri
 	}
 	return nil
 }
+
+// RemoveUnknownFields removes those fields from the object which are not specified in the properties construct of the schema.
+func RemoveUnknownFields(object map[string]interface{}, schema *apiextensions.JSONSchemaProps) {
+	if object == nil || schema == nil || schema.Properties == nil {
+		return
+	}
+
+	for field := range object {
+		// don't consider typemeta and objectmeta
+		if field == "apiVersion" || field == "kind" || field == "metadata" {
+			continue
+		}
+
+		// TODO: patternProperties as well?
+		pSchema, ok := schema.Properties[field]
+
+		// if the field does not exist in the properties construct, remove it
+		if !ok {
+			delete(object, field)
+			continue
+		}
+
+		// check if the subobject is a map
+		subObject := object[field]
+		subObjectMap, ok := subObject.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// if the subObject is a map, then the subschema should have a properties construct
+		// if it doesn't, it means we can delete all fields in it and hence the field itself
+		// TODO: double check if we want this behavior?
```

Comment:
This will not just break what I do, it will not leave me any choice at all. I need a way to describe a `map[string]string` in schema to describe annotations and labels in an object that is encoded in my CR. https://github.com/kubernetes/kubernetes/issues/59485

Reply:
We need a way to allow `map[string]string`, actually even `map[string]interface{}`. The question is how to express this with JSON Schema.

One way would be: 
- defined `additionalSchema` => drop unknown fields
- non-empty `patternSchema` => it's a `map[string]...`.

At least one of them must be defined (as now with `additionalSchema` alone).

Reply:
https://datatracker.ietf.org/doc/draft-handrews-json-schema-validation/?include_text=1

Reply:
cross-link https://github.com/kubernetes/kubernetes/issues/59485